docs/references/unittest.rst

```diff
@@ -0,0 +1,179 @@
+Unit Test Framework
+========================
+
+.. note::
+	Available in v0.11.0 and later
+
+Overview
+*********
+
+You can write test cases for your tool by defining the ``tests()`` function in your Tool class.
+Test cases defined by this function will be picked up by ``janisdk run-test`` command.
+
+This unit test framework provides several predefined preprocessors to
+transform Janis execution output data in the format that can be tested.
+For example, there is a preprocessor to read the md5 checksum of an output file.
+In addition to the predefined preprocessors, the frameworks also allows users to define and pass their own preprocessors.
+
+
+Define test cases
+-----------------
+You can define multiple test cases per tool.
+For each test case, you can declare multiple expected outputs.
+Mostly, you really only need to define more test cases if they require different input data.
+
+
+.. code-block:: python
+
+    class BwaAligner(BioinformaticsWorkflow):
+        def id(self):
+            return "BwaAligner"
+
+        ...
+
+        def tests(self):
+            return [
+                TTestCase(
+                    name="basic",
+                    input={
+                        "bam": "https://some-public-container/directory/small.bam"
+                    },
+                    output=[
+                        TTestExpectedOutput(
+                            tag="out",
+                            preprocessor=TTestPreprocessor.FileMd5,
+                            operator=operator.eq,
+                            expected_value="dc58fe92a9bb0c897c85804758dfadbf",
+                        ),
+                        TTestExpectedOutput(
+                            tag="out",
+                            preprocessor=TTestPreprocessor.FileContent,
+                            operator=operator.contains,
+                            expected_value="19384 + 0 in total (QC-passed reads + QC-failed reads)",
+                        ),
+                        TTestExpectedOutput(
+                            tag="out",
+                            preprocessor=TTestPreprocessor.LineCount,
+                            operator=operator.eq,
+                            expected_value=13,
+                        ),
+                    ],
+                )
+            ]
+
+Run the tests
+-------------
+.. code-block:: console
+
+    # Run a specific test case
+    janisdk run-test --test-case [TEST CASE NAME] [TOOL ID]
+
+    # Run ALL test cases of one tool
+    janisdk run-test [TOOL ID]
+
+To run the example test case shown above:
+
+.. code-block:: console
+
+    janisdk run-test --test-case=basic BwaAligner
+
```

Comment:
Can we include a one-liner here for running all tests from all tools?

Reply:
makes sense! just added :)